extensions/llama/python/llama.py

```diff
@@ -83,21 +83,31 @@ async def _run_inference_helper(self, model_input, options) -> ExecuteResult:
                 if options:
                     options.stream_callback(data, acc, index)
             print(flush=True)
-            return ExecuteResult(output_type="execute_result", data=[acc], metadata={})
+            return ExecuteResult(
+                output_type="execute_result",
+                data=acc,
+                metadata={}
+            )
         else:
             response = llm(model_input)
             try:
                 texts = [r["choices"][0]["text"] for r in response]
             except TypeError:
                 texts = [response["choices"][0]["text"]]
 
-            return ExecuteResult(output_type="execute_result", data=texts, metadata={})
+            return ExecuteResult(
+                output_type="execute_result",
+                data=texts[0],
```

Comment:
I'm not sure about this part -- what about the rest of the items in the `texts` array?

Reply:
This is a good callout. I dug deeper and realized that this extension stores all the data under a single prompt output, vs out other chat-bot models which stores each message as it's own individual prompt. Ex: https://github.com/lastmile-ai/aiconfig/blob/main/python/src/aiconfig/default_parsers/openai.py#L93-L130

For now I'll use `texts[-1]` to ensure we get the most recent conversation message, but in general I think this extension (and associated cookbook) will need to be refactored to match our other ways of storing message history as 1-1 mapping to prompt. Llama also supports chat completion params so this shouldn't take too long: https://github.com/abetlen/llama-cpp-python/blob/main/llama_cpp/llama.py#L791-L796

I made GH issue for it in